runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java

```diff
@@ -124,6 +130,9 @@
   private transient List<
```

Comment:
I think we don't need `restoredState` and the logic that does the copying in `initializeState()`. We should be able to just use `stateForCheckpoint` in `open()` for initialisation.